### DISCUSSION TITLE
add build-variants app crashed

### DISCUSSION THREAD

**Discussion by armanhossiny** [axmolengine/axmol#1571] [Q&A] [answered]
hi
I want add build-variants to Gradle Project but app crashed, problem is not  gradle, I think bug axmol.
and test projcet cocos2dx v4 and project android studio with this code and worked.

I use this link:
https://developer.android.com/reference/tools/gradle-api/8.1/com/android/build/api/dsl/ProductFlavor

code:
```
android {
        flavorDimensions 'tier'

        productFlavors {

            paid {
                dimension 'tier'
            }

            free {
                dimension 'tier'
            }
        }
    }
```
when add this code to Gradle project HelloWorld, app crashed with this error:
list error in Logcat Android studio:
1
![1](https://github.com/axmolengine/axmol/assets/42970068/432960ec-cb9d-4f57-8c65-83b507930a05)
2
![2](https://github.com/axmolengine/axmol/assets/42970068/ca065c54-f381-4232-a91b-3f47c87b4920)
3
![3](https://github.com/axmolengine/axmol/assets/42970068/fde15492-f7db-49b6-a6e0-cf0c9ea81767)

gradle:
![4](https://github.com/axmolengine/axmol/assets/42970068/fd05c514-064b-44ec-ba41-f6085d3ff09a)

**Comment by rh101** [comment, marked as answer]
Check this thread: https://github.com/axmolengine/axmol/discussions/1508

**Reply by armanhossiny** [reply]
thank you, but this complex, I think better if remove in new version axmol.   :)

**Reply by rh101** [reply]
It's been updated in this commit:

https://github.com/axmolengine/axmol/commit/dd7c9deaf9d0d837261ca6c43cba692474442be2

**Reply by armanhossiny** [reply]
thank you, this is great
